src/plugins/dashboard/common/dashboard_container/types.ts

```diff
@@ -53,6 +53,7 @@ export interface DashboardContainerInput extends EmbeddableInput {
   timeRange?: TimeRange;
   timeslice?: [number, number];
   refreshInterval?: RefreshInterval;
+  fullScreenMode?: boolean;
```

Comment:
I think this should be part of the `InitialComponentState` type (defined in `src/plugins/dashboard/public/dashboard_api/get_dashboard_api.ts`) rather than `DashboardContainerInput`. The input time is used to reflect things that are **saved** into the dashboard - and this should not be saved. 

Then, in `getDashboardApi`, we should initialize the behaviour subject to the correct value and use that as our source of truth. Please let me know if you have questions :bow: 

cc @nreese

Reply:
yeah, I'll try to fix it

Reply:
I have a question, the `fullscreenMode` value is get from the `getCreationOptions.getInitialInput` function, which is defined in `src/plugins/dashboard/public/dashboard_app/dashboard_app.tsx`. And `SavedDashboardInput` is the type of the return value of `getInitialInput`, which is defined like this
```
export type SavedDashboardInput = DashboardContainerInput & {
  /**
   * Serialized control group state.
   * Contains state loaded from dashboard saved object
   */
  controlGroupInput?: DashboardAttributes['controlGroupInput'] | undefined;
  /**
   * Runtime control group state.
   * Contains state passed from dashboard locator
   * Use runtime state when building input for portable dashboards
   */
  controlGroupState?: Partial<ControlGroupRuntimeState>;
};
```
So should I change the the implementation of the `getInitialInput` function to get the `fullscreenMode` value in a another function?

Reply:
You'll want to take a look at `src/plugins/dashboard/public/dashboard_container/embeddable/create/create_dashboard.ts` where we define `InitialComponentState`:

https://github.com/elastic/kibana/blob/2a786b88a131472c349f0bda431efe3ed6fbeaf6/src/plugins/dashboard/public/dashboard_container/embeddable/create/create_dashboard.ts#L104-L123

From my view, I think we should handle `fullscreenMode` in the same way we handle `isEmbeddedExternally` - so, because of this, I think it is reasonable to add `fullscreenMode` to the `DashboardCreationOptions` in this case.

Reply:
Yes, you're right. But the `fullscreenMode` value is get from the function `loadAndRemoveDashboardState`

https://github.com/elastic/kibana/blob/main/src/plugins/dashboard/public/dashboard_app/url/url_utils.ts#L67-L89

`DashboardContainerInput` is the type of return value of `loadAndRemoveDashboardState`, and there is no define of `fullscreenMode`

https://github.com/elastic/kibana/blob/main/src/plugins/dashboard/common/dashboard_container/types.ts#L48-L73

I can add `fullscreenMode` to the `DashboardCreationOptions`, but the `DashboardContainerInput` should be changed together.

Reply:
@Heenawter I submitted a new changes, can you take a look.